index.js

```diff
@@ -143,18 +156,17 @@ const parseArgs = (
           StringPrototypeSlice(arg, 0, index),
           StringPrototypeSlice(arg, index + 1),
           result);
-      } else if (pos + 1 < argv.length &&
-        !StringPrototypeStartsWith(argv[pos + 1], '-')
-      ) {
+      } else if (peek(argv, pos) !== TOKENS.FLAG) {
```

Comment:
Processing the last arg will now drop into this conditional, whereas previously it would drop through to the next test.

i.e. removed `pos + 1 < argv.length` test

Reply:
This block is processing an option with a following argument available as a value, if needed.

Reply:
The bounds check was moved into peek, if we decide not to handle negative numbers I'll abandon this refactor any ways.